File: strategy/php/flynoway.php
<?php
/**
 * File flynoway.php
 *
 * @author kujaomega
 */

/**
 * This is the flying behavior implementation for ducks that don't fly.
 */
class FlyNoWay implements FlyBehavior
{
    /**
     * This is the fly method.
     */
    public function fly()
    {
        echo( 'I can\'t fly!');
    }
}

File: strategy/java/FlyNoWay.java
/*
 * File FlyBehavior.java
 * 
 * @author kujaomega
 */

/**
 * This class will set the impossibility to fly.
 * 
 * Class FlyNoWay
 */
public class FlyNoWay implements FlyBehavior{
    
    /**
     * This is the fly method.
     */
    public void fly(){
        System.out.println("I can't fly");
    }
}
